aws/code/HPOConfig.py

```diff
@@ -0,0 +1,228 @@
+                compute_type = 'multi-CPU'
+            elif 'singlecpu' in compute_selection:
+                compute_type = 'single-CPU'
+            elif 'singlegpu' in compute_selection:
+                compute_type = 'single-GPU'
+
+            # parse CV folds
+            cv_folds = int(os.environ['CV_FOLDS'])
+
+        except KeyError as error:
+            hpo_log.info(f'Configuration parser failed : {error}')
+
+        assert (dataset_type in ['Airline', 'NYCTaxi', 'BYOData'])
+        assert (model_type in ['RandomForest', 'XGBoost'])
+        assert (compute_type in ['single-GPU', 'multi-GPU',
+                                 'single-CPU', 'multi-CPU'])
+        assert (cv_folds >= 1)
+
+        hpo_log.info(f'  Dataset: {dataset_type}\n'
+                     f'  Compute: {compute_type}\n'
+                     f'  Algorithm: {model_type}\n'
+                     f'  CV_folds: {cv_folds}\n')
+
+        return dataset_type, model_type, compute_type, cv_folds
+
+    def parse_hyper_parameter_inputs(self, input_args):
+        """ Parse hyperparmeters provided by the HPO orchestrator """
+        hpo_log.info('parsing model hyperparameters from command line arguments...log')
+        parser = argparse.ArgumentParser()
+
+        if 'XGBoost' in self.model_type:
+            # intentionally breaking PEP8 below for argument alignment
+            parser.add_argument( '--max_depth',       type = int,   default = 5 )
+            parser.add_argument( '--num_boost_round', type = int,   default = 10 )
+            parser.add_argument( '--subsample',       type = float, default = .9 )
+            parser.add_argument( '--learning_rate',   type = float, default = 0.3)
+            parser.add_argument( '--reg_lambda',      type = float, default = 1)
+            parser.add_argument( '--gamma',           type = float, default = 0. )
+            parser.add_argument( '--alpha',           type = float, default = 0. )
+            parser.add_argument( '--seed',            type = int,   default = 0 )
+
+            args, unknown_args = parser.parse_known_args(input_args)
+
+            model_params = {
+                'max_depth': args.max_depth,
+                'num_boost_round': args.num_boost_round,
+                'learning_rate': args.learning_rate,
+                'gamma': args.gamma,
+                'lambda': args.reg_lambda,
+                'random_state': args.seed,
+                'verbosity': 0,
+                'seed': args.seed,
+                'objective': 'binary:logistic'
+            }
+
+            if 'single-CPU' in self.compute_type:
+                model_params.update({'nthreads': os.cpu_count()})
+
+            if 'GPU' in self.compute_type:
+                model_params.update({'tree_method': 'gpu_hist'})
+            else:
+                model_params.update({'tree_method': 'hist'})
+
+        elif 'RandomForest' in self.model_type:
+            # intentionally breaking PEP8 below for argument alignment
```

Comment:
Generally, this is marked with `noqa` directives, since it signals the intention and prevents further linting alarms. No need to change it for this PR; just a note for the future.